### PR TITLE
Chore/replace jenkins build

### PR DESCRIPTION
**Pull request by 425devon**
Replace Jenkins file with GHA CI

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -3,10 +3,102 @@ concurrency:
...
+        uses: actions/checkout@v3
+
+      - name: checkout kong portal tools
+        uses: actions/checkout@v3
+        with:
+          repository: Kong/kong-portal-tools
+          path: kong-portal-tools
+          token: ${{ secrets.GHA_KONG_BOT_READ_TOKEN }}
+
+      - name: copy kong-portal-tools/kong-portal-ci to current dir
```
Comment (zekth) [inline .github/workflows/ci.yml:29]: What's the reason to copy files instead of cd in the directory and doing a make build?

Comment (425devon) [reply .github/workflows/ci.yml:29]: matter of preference I suppose, I could set `working-directory:` for each make step ::shrug:: this is 1:1 with the previous flow using "bootstrap.mk"

Comment (425devon) [reply .github/workflows/ci.yml:29]: which reminds me that file can go, do you have a compelling reason to use  `working-directory:`?

Comment (zekth) [reply .github/workflows/ci.yml:29]: `working-directory` seems to be the right call IMO

Comment (nateslo) [reply .github/workflows/ci.yml:29]: I vote that we maintain the existing setup and get this working. Can always revisit and optimize it later

**Review by nateslo — APPROVED** [review]

**Pull request merged**